packages/cli/src/UserManagement/Interfaces.ts

```diff
@@ -47,3 +52,22 @@ export declare namespace UpdateSelfRequest {
 	export type Password = AuthenticatedRequest<UpdateSelfPayload.Password>;
 	export type SurveyAnswers = AuthenticatedRequest<UpdateSelfPayload.SurveyAnswers>;
 }
+
+// ----------------------------------
+//      password reset requests
+// ----------------------------------
+
+declare namespace PasswordResetPayload {
+	type Email = Pick<PublicUserData, 'email'>;
+	type NewPassword = Pick<PublicUserData, 'password'> & { token?: string; id?: string };
+}
+
+declare namespace PasswordResetQuery {
+	type Credentials = { u?: string; t?: string };
```

Comment:
This is inside of a namespace so only referrable by its full name as `PasswordResetQuery.Credentials`.

Reply:
Both params are required for the check to succeed, but at the type level, both params are optional because the requester may forget to send them, so TS requires us to check for them.

Reply:
alright thanks, just synced with Omar and we decided to not obscure the params for now, sorry for additional efforts